pkg/kubelet/config/file_linux.go

```diff
@@ -57,30 +83,25 @@ func (s *sourceFile) watch() error {
 	}
 	defer w.Close()
 
-	err = w.AddWatch(s.path, inotify.IN_DELETE_SELF|inotify.IN_CREATE|inotify.IN_MOVED_TO|inotify.IN_MODIFY|inotify.IN_MOVED_FROM|inotify.IN_DELETE)
+	err = w.AddWatch(s.path, inotify.IN_DELETE_SELF|inotify.IN_CREATE|inotify.IN_MOVED_TO|inotify.IN_MODIFY|inotify.IN_MOVED_FROM|inotify.IN_DELETE|inotify.IN_ATTRIB)
 	if err != nil {
 		return fmt.Errorf("unable to create inotify for path %q: %v", s.path, err)
 	}
 
-	// Reset store with manifest files already existing when starting
-	if err := s.resetStoreFromPath(); err != nil {
-		return fmt.Errorf("unable to read manifest path %q: %v", s.path, err)
-	}
-
 	for {
 		select {
 		case event := <-w.Event:
-			err = s.processEvent(event)
+			err = s.produceWatchEvent(event)
 			if err != nil {
-				return fmt.Errorf("error while processing event (%+v): %v", event, err)
+				glog.Errorf("error while processing inotify event (%+v): %v", event, err)
```

Comment:
It looks like the only type of error that can occur here is a delete self. In that case, don't you want to break out of this loop and try to recreate the watcher?